lnrpc/walletrpc/walletkit.proto

```diff
@@ -80,23 +80,45 @@ service WalletKit {
     rpc ListAddresses (ListAddressesRequest) returns (ListAddressesResponse);
 
     /*
-    SignMessageWithAddr returns a signature with the private key of the provided
-    address. The address lookup in the wallet will consider the default account
-    but can be changed with a cmd argument. In addition the address type is
-    needed to safe type and not check the address for all possible types. In
-    case the address type and the address do not match, no other types are
-    considered
+    SignMessageWithAddr returns the compact signature (base64 encoded) created
+    with the private key of the provided address. This requireds the address
+    to be solely based on a public key lock (no scripts). Obviously the internal
+    lnd wallet has to possess the private key of the address otherwise
+    an error is returned.
+
+    This method aims to provide full compatibility with the bitcoin-core and
+    btcd implementation. Bitcoin-core's algorithm is not specified in a
+    BIP and only applicable for legacy addresses. This method enhances the
+    signing for additional address types: P2WKH, NP2PKH, P2TR.
+    For P2TR addresses this represents a special case. Edsca is used to create
+    a compact signature which makes the public key of the signature recoverable.
+    For Schnorr no known compact signing algorithm exists yet. This could change
+    in the future, one could use the external public key for P2TR but it was
+    decided to use the internal private key to sign the message.
```

Comment:
The decision process shouldn't really be reflected in the API docs. I think this should just reflect how it is currently implemented and be changed if the behavior is different.

Also, ECDSA should be all uppercase IMO.

Reply:
Agree, I took it out but still left the comment that for P2TR its a special case that we use ECDSA